env0/data_custom_roles.go

```diff
@@ -0,0 +1,51 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataCustomRoles() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataCustomRolesRead,
+
+		Schema: map[string]*schema.Schema{
+			"names": {
+				Type:        schema.TypeList,
+				Description: "list of all custom roles (by name)",
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type:        schema.TypeString,
+					Description: "the role name",
+				},
+			},
+		},
+	}
+}
+
+func dataCustomRolesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+	roles, err := apiClient.Roles()
+	if err != nil {
+		return diag.Errorf("Failed to get custom roles: %v", err)
+	}
+
+	data := []string{}
+
+	for _, role := range roles {
+		// Built-in roles do not have an identifier.
```

Comment:
If this ever changes, it will break.
There is no other way to differentiate between custom and pre-built.
At the moment both are returned via the API.

Reply:
actually we have a isDefaultRole on the response for that api which we omitted from the api, the omission might be a mistake on our end, I asked in our rbac channel and will update here 👍

Reply:
so there is actually no reason for the `isDefaultRole` omission on the api so ill just open a pr to add it to it, but I am pretty sure can use it from the response already 🙂

Reply:
@ItamarMalka - thanks for the feedback.
I have fixed it.